crowdsourcehinter/static/js/src/crowdsourcehinter.js

```diff
@@ -0,0 +1,305 @@
+     * @param hint is the first hint that was shown to the student
+     * @param student_answer is the first incorrect answer submitted by the student
+     */
+    function showHintFeedback(hint, student_answer){
+        $(".csh_student_answer", element).each(function(){
+            if ($(this).find('.csh_answer_text').attr('answer') == student_answer){
+                var html = "";
+                $(function(){
+                    var data = {
+                        hint: hint
+                    };
+                    html = Mustache.render($("#show_hint_feedback").html(), data);
+                });
+                $(this).append(html);
+                var html = "";
+                var template = $('#add_hint_creation').html();
+                var data = {};
+                html = Mustache.render(template, data);
+                $(this).append(html);
+            }
+        });
+    }
+
+    /**
+     * Show options to remove or return reported hints from/to the hint pool. Called after
+     * correctly answering the question, only visible to staff.
+     * @param result is the reported hint text
+     */
+    function showReportedFeedback(result){
+        var html = "";
+        $(function(){
+            var template = $('#show_reported_feedback').html();
+            var data = {
+                hint: result
+            };
+            html = Mustache.render(template, data);
+        });
+        $(".csh_reported_hints", element).append(html);
+    }
+
+    /**
+     * Append new divisions into html for each answer the student submitted before correctly 
+     * answering the question. showHintFeedback appends new hints into these divs.
+     * When the hinter is set to show best, only one div will be created.
+     * @param student_answers is the text of the student's incorrect answer
+     */
+    function showStudentSubmissionHistory(student_answers){
+        var html = "";
+        var template = $('#show_answer_feedback').html();
+        var data = {
+            answer: student_answers
+        };
+        html = Mustache.render(template, data);
+        $(".csh_feedback", element).append(html);
+    }
+
+    /**
+     * Set up student feedback on hints and contribution of new hints. Incorrect answer(s) and the
+     * the corresponding hint(s) shown to the student are displayed. Students can upvote/downvote/report
+     * hints or contribute a new hint for their incorrect answer(s).
+     * Only one incorrect answer and hint will be shown when the hinter is set to show best.
+     * @param result is a dictionary of incorrect answers and hints, with the index being the hint and the value
+     * being the incorrect answer
+     */
+    function showStudentContribution(result){
+    //Set up the student feedback stage. Each student answer and all answer-specific hints for that answer are shown
+    //to the student, as well as an option to create a new hint for an answer.
+        if(data.isStaff){
```

Comment:
if (data.isStaff) { // brief description at a product level of what this does. E.g. "Show an interface for moderating flagged hints" or "Show an advanced settings" or whatever the code below causes to happen.